### PR TITLE
Set HTTP proxy settings if they exist

### PR DESCRIPTION
**Pull request by gastaldi**
May fix https://github.com/quarkusio/quarkus/discussions/44802#discussioncomment-11406493
(Waiting for the OP's confirmation)

### CODE REVIEW

**Review by zhfeng — APPROVED** [review]
LGTM

File: core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java [hunk 1]
```diff
@@ -339,6 +339,18 @@ protected void configure() {
             if (port > 0) {
                 addFixedExposedPort(port, ARTEMIS_PORT);
             }
+            String httpProxy = System.getenv("HTTP_PROXY");
```
Comment (middagj) [inline core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java:342]: Perhaps reducing the logic using (or something similar)
```java
  withSystemEnv("HTTP_PROXY")
  withSystemEnv("HTTPS_PROXY")
  withSystemEnv("NO_PROXY")
}

private void withSystemEnv(String name) {
  val value = System.getenv(name)
  if (value != null) {
    withEnv(name, value)
  }
}
```

**Review by middagj — APPROVED** [review]

**Review by turing85 — APPROVED** [review]

**Comment by gastaldi** [comment]
3.15 perhaps, 3.8 not sure. But I'm interested in knowing if that really fixes the problem first

**Comment by turing85** [comment]
Closed since https://github.com/quarkusio/quarkus/discussions/44802 has been closed.

**Pull request closed without merging**